---
title: "Ant1-KO Differential Expression"
author: "Jeremy Leipzig"
date: "10/22/2015"
output:
  html_document:
    toc: true
---







```
## Error in rename(., ensembl_gene_id = id): unused argument (ensembl_gene_id = id)
```


```
## Error in getBM(attributes = c("ensembl_gene_id", "external_gene_name", : object 'testres' not found
```

```
## Error in getBM(attributes = c("ensembl_gene_id", "entrezgene"), filters = "ensembl_gene_id", : object 'testres' not found
```

```
## Error in paste(values, "", collapse = ",", sep = ""): object 'anno_entrez' not found
```

```
## Error in rename(., entrez.human = c2.geneId, cardiomyopathy_score = c0.score): unused arguments (entrez.human = c2.geneId, cardiomyopathy_score = c0.score)
```

```
## Error in eval(expr, envir, enclos): object 'anno_entrez' not found
```

```
## Error in is.data.frame(x): object 'res' not found
```

```
## Error in save(res, file = paste(resfile, "RData", sep = "."), compress = TRUE): object 'res' not found
```

```
## Error in empty(.data): object 'entrez' not found
```

```
## Error in eval(expr, envir, enclos): object 'unique_entrez' not found
```

```
## Error in eval(expr, envir, enclos): object 'unique_entrez' not found
```

```
## Error in names(genego)[1] <- "entrez": object 'genego' not found
```

```
## Error in subset(genego[, c("entrez", "foldChange", "padj")], padj < 0.05 & : error in evaluating the argument 'x' in selecting a method for function 'subset': Error: object 'genego' not found
```

```
## Error in is.data.frame(x): object 'genego_sig' not found
```

# Introduction
## Samples
Below are results of the STAR mapping. Conditions are M for Muscle, H for Heart, T for Treatment(ANT1), N for Normal (B6ME).
<!--html_preserve--><div id="htmlwidget-8998" style="width:100%;height:auto;" class="datatables"></div>
<script type="application/json" data-for="htmlwidget-8998">{"x":{"data":[["IonXpressRNA_002.R_2013_11_26_13_55_09_user_1PR-8-RNA-Seq_whole_transcriptome","IonXpressRNA_004.R_2013_11_26_20_48_53_user_1PR-9-RNA-Seq_whole_transcriptome","IonXpressRNA_006.R_2013_12_04_09_37_33_1PR-10-RNA-Seq_whole_transcriptome_76303","IonXpressRNA_008.R_2013_12_06_12_45_12_user_1PR-11-RNA-Seq_whole_transcriptome","IonXpressRNA_001.R_2013_11_26_13_55_09_user_1PR-8-RNA-Seq_whole_transcriptome","IonXpressRNA_003.R_2013_11_26_20_48_53_user_1PR-9-RNA-Seq_whole_transcriptome","IonXpressRNA_005.R_2013_12_04_09_37_33_1PR-10-RNA-Seq_whole_transcriptome_76303","IonXpressRNA_007.R_2013_12_06_12_45_12_user_1PR-11-RNA-Seq_whole_transcriptome","IonXpressRNA_010.R_2013_12_19_16_11_21_user_1PR-13-RNA-Seq_whole_transcriptome","IonXpressRNA_012.R_2013_12_18_20_25_40_user_1PR-12-RNA-Seq_whole_transcriptome","IonXpressRNA_014.R_2013_12_20_12_50_23_user_1PR-14-RNA-Seq_whole_transcriptome","IonXpressRNA_016.R_2013_12_21_20_34_59_user_1PR-15-RNA-Seq_whole_transcriptome","IonXpressRNA_009.R_2013_12_19_16_11_21_user_1PR-13-RNA-Seq_whole_transcriptome","IonXpressRNA_011.R_2013_12_18_20_25_40_user_1PR-12-RNA-Seq_whole_transcriptome","IonXpressRNA_013.R_2013_12_20_12_50_23_user_1PR-14-RNA-Seq_whole_transcriptome","IonXpressRNA_015.R_2013_12_21_20_34_59_user_1PR-15-RNA-Seq_whole_transcriptome"],["IonXpressRNA_002...","IonXpressRNA_004...","IonXpressRNA_006...","IonXpressRNA_008...","IonXpressRNA_001...","IonXpressRNA_003...","IonXpressRNA_005...","IonXpressRNA_007...","IonXpressRNA_010...","IonXpressRNA_012...","IonXpressRNA_014...","IonXpressRNA_016...","IonXpressRNA_009...","IonXpressRNA_011...","IonXpressRNA_013...","IonXpressRNA_015..."],[25482089,31826234,29907063,34668896,35471247,28749689,27702368,15598072,20270224,26586149,32947023,28872835,33005189,28163291,27656576,39656083],["44.12%","50.70%","51.24%","50.78%","48.76%","48.81%","50.68%","59.85%","42.32%","38.02%","51.75%","64.05%","40.90%","49.28%","38.41%","47.51%"],["17.41%","19.75%","14.94%","19.78%","18.60%","18.49%","16.09%","14.62%","19.61%","22.20%","18.60%","14.46%","20.21%","22.31%","19.40%","18.79%"],["MT","MT","MT","MT","MN","MN","MN","MN","HT","HT","HT","HT","HN","HN","HN","HN"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> </th>\n      <th>Sample</th>\n      <th>Input.reads</th>\n      <th>Uniquely.mapped</th>\n      <th>Mapped.to.multiple.loci</th>\n      <th>cond</th>\n    </tr>\n  </thead>\n</table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"caption":"<caption>STAR mapping results</caption>","filter":"none"},"evals":[]}</script><!--/html_preserve-->

# Alignment
After trimming, transcript counts were generated by a STAR alignment to genomic sequences.

The STAR  used for the samples is as follows:
```
{STAR} --genomeDir {STARREFDIR} --outFileNamePrefix {wildcards.sample}_ --readFilesIn {input} --runThreadN 24 
--genomeLoad NoSharedMemory --outSAMattributes All --outSAMstrandField intronMotif --sjdbGTFfile {GTFFILE}
```

## Normalization
ERCC spike-ins were downloaded from [here](http://tools.invitrogen.com/downloads/ERCC92.fa) and used as a reference. Total counts from all 92 sequences are below. These are used for the *sizeFactors* in DESeq such that they become divisors for each sample count.
<!--html_preserve--><div id="htmlwidget-997" style="width:100%;height:auto;" class="datatables"></div>
<script type="application/json" data-for="htmlwidget-997">{"x":{"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["IonXpressRNA_002.R_2013_11_26_13_55_09_user_1PR-8-RNA-Seq_whole_transcriptome","IonXpressRNA_004.R_2013_11_26_20_48_53_user_1PR-9-RNA-Seq_whole_transcriptome","IonXpressRNA_006.R_2013_12_04_09_37_33_1PR-10-RNA-Seq_whole_transcriptome_76303","IonXpressRNA_008.R_2013_12_06_12_45_12_user_1PR-11-RNA-Seq_whole_transcriptome","IonXpressRNA_001.R_2013_11_26_13_55_09_user_1PR-8-RNA-Seq_whole_transcriptome","IonXpressRNA_003.R_2013_11_26_20_48_53_user_1PR-9-RNA-Seq_whole_transcriptome","IonXpressRNA_005.R_2013_12_04_09_37_33_1PR-10-RNA-Seq_whole_transcriptome_76303","IonXpressRNA_007.R_2013_12_06_12_45_12_user_1PR-11-RNA-Seq_whole_transcriptome","IonXpressRNA_010.R_2013_12_19_16_11_21_user_1PR-13-RNA-Seq_whole_transcriptome","IonXpressRNA_012.R_2013_12_18_20_25_40_user_1PR-12-RNA-Seq_whole_transcriptome","IonXpressRNA_014.R_2013_12_20_12_50_23_user_1PR-14-RNA-Seq_whole_transcriptome","IonXpressRNA_016.R_2013_12_21_20_34_59_user_1PR-15-RNA-Seq_whole_transcriptome","IonXpressRNA_009.R_2013_12_19_16_11_21_user_1PR-13-RNA-Seq_whole_transcriptome","IonXpressRNA_011.R_2013_12_18_20_25_40_user_1PR-12-RNA-Seq_whole_transcriptome","IonXpressRNA_013.R_2013_12_20_12_50_23_user_1PR-14-RNA-Seq_whole_transcriptome","IonXpressRNA_015.R_2013_12_21_20_34_59_user_1PR-15-RNA-Seq_whole_transcriptome"],[25216131,31454860,29509934,34313656,34958364,28273488,27318394,15401398,20047401,26273728,32544990,28730069,32647281,27830405,27382176,39224079]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> </th>\n      <th>sample</th>\n      <th>count</th>\n    </tr>\n  </thead>\n</table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"caption":"<caption>Total ERCC read counts</caption>","filter":"none"},"evals":[]}</script><!--/html_preserve-->


# DESeq
A reanalysis of the non-normalized count data was performed using the
R-based RNA-Seq analysis package DESeq [Anders](http://dx.doi.org/10.1038/npre.2010.4282.2) in order to
detect differential expression between normal and treatment.

This report uses DESeq version 1.20.0.

DESeq offers different statistical models and parameters to choose from depending on the data to be analyzed.

Empirical dispersion can be computed using a *pooled*, *per-condition*, or *blind* (no replicates) method.

The following, sharing modes, which determine how much information is used to inform the dispersion of other genes, are available:

* `fit-only` is appropriate for only a few replicates
* `maximum` is more conservative for three or four replicates, 
* `gene-est-only` is more aggressive, and is ideal for many replicates


For this report, we have used the per-condition method and maximum sharing mode.

Only genes with a combined count of 100 among all the samples (both conditions and tissues)
14642 hits fit this criteria.

# Muscle results
## Significantly DE genes in muscle
Significantly differentially expressed genes (padj<0.01). Fold change is ANT1/B6ME.


```
## Error in eval(expr, envir, enclos): object 'res' not found
```

```
## Error in base::rownames(data): object 'resSig' not found
```

## Upregulated genes in muscle

```
## Error in head(subset(resSig[order(-resSig$fc), ], fc > 1), n = 50): error in evaluating the argument 'x' in selecting a method for function 'head': Error in subset(resSig[order(-resSig$fc), ], fc > 1) : 
##   error in evaluating the argument 'x' in selecting a method for function 'subset': Error: object 'resSig' not found
```

```
## Error in nrow(upRegs): error in evaluating the argument 'x' in selecting a method for function 'nrow': Error: object 'upRegs' not found
```


## Downregulated genes in muscle

```
## Error in head(subset(resSig[order(resSig$fc), ], fc <= 1), n = 50): error in evaluating the argument 'x' in selecting a method for function 'head': Error in subset(resSig[order(resSig$fc), ], fc <= 1) : 
##   error in evaluating the argument 'x' in selecting a method for function 'subset': Error: object 'resSig' not found
```

```
## Error in nrow(downRegs): error in evaluating the argument 'x' in selecting a method for function 'nrow': Error: object 'downRegs' not found
```

## Fold change and significance muscle
Genes with high fold change may not be significantly differentially expressed simply due to high variance.


```
## Error in ggplot(res, aes(x = log2(baseMean), y = log2FoldChange)): object 'res' not found
```

# Heart results
## Significantly DE genes in heart

```
## Error in rename(., ensembl_gene_id = id): unused argument (ensembl_gene_id = id)
```


```r
#http://www.bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/biomaRt.pdf
mart <- useMart(biomart = "ensembl", dataset = "mmusculus_gene_ensembl")

anno_entrez <- getBM(attributes = c("ensembl_gene_id","external_gene_name","entrezgene"),
                 filters = "ensembl_gene_id", values = testres$ensembl_gene_id,
                 mart = mart)
```

```
## Error in getBM(attributes = c("ensembl_gene_id", "external_gene_name", : object 'testres' not found
```

```r
entrez <- getBM(attributes = c("ensembl_gene_id","entrezgene"),
                 filters = "ensembl_gene_id", values = testres$ensembl_gene_id,
                 mart = mart)
```

```
## Error in getBM(attributes = c("ensembl_gene_id", "entrezgene"), filters = "ensembl_gene_id", : object 'testres' not found
```

```r
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")

getLDS(attributesL = c("entrezgene"),
       filters = "entrezgene", values = anno_entrez$entrezgene, mart = mart,
       attributes = c("entrezgene"), martL = human) %>% rename(entrezgene=EntrezGene.ID,entrez.human=EntrezGene.ID.1) -> homologs
```

```
## Error in paste(values, "", collapse = ",", sep = ""): object 'anno_entrez' not found
```

```r
get(load("cardiomyopathy.RData")) %>% rename(entrez.human=c2.geneId,cardiomyopathy_score=c0.score)->cmyo_genes
```

```
## Error in rename(., entrez.human = c2.geneId, cardiomyopathy_score = c0.score): unused arguments (entrez.human = c2.geneId, cardiomyopathy_score = c0.score)
```

```r
#join to cardiomyopathy tables using homologues, remove dups
anno_entrez %>% 
  join(homologs,by="entrezgene",type="left", match = "first") %>% 
  join(cmyo_genes,by="entrez.human",type="left", match = "first") %>% 
  group_by(ensembl_gene_id) %>% 
  filter(row_number() == 1) %>% 
  join(testres, by="ensembl_gene_id", type="right") %>% 
  rename(gene=external_gene_name) -> res
```

```
## Error in eval(expr, envir, enclos): object 'anno_entrez' not found
```

```r
write.csv(res,file=resfile)
```

```
## Error in is.data.frame(x): object 'res' not found
```

```r
save(res,file=paste(resfile,"RData",sep="."),compress=TRUE)
```

```
## Error in save(res, file = paste(resfile, "RData", sep = "."), compress = TRUE): object 'res' not found
```

```r
#genego convenience

unique_entrez<-ddply(entrez,.(ensembl_gene_id),function(dt){dt[1,]})
```

```
## Error in empty(.data): object 'entrez' not found
```

```r
row.names(unique_entrez)<-unique_entrez$ensembl_gene_id
```

```
## Error in eval(expr, envir, enclos): object 'unique_entrez' not found
```

```r
genego<-cbind(unique_entrez[res$gene,"entrezgene"],testres)
```

```
## Error in eval(expr, envir, enclos): object 'unique_entrez' not found
```

```r
names(genego)[1]<-"entrez"
```

```
## Error in names(genego)[1] <- "entrez": object 'genego' not found
```

```r
genego_sig<-subset(genego[,c("entrez","foldChange","padj")],padj < 0.05 & !is.na(entrez))
```

```
## Error in subset(genego[, c("entrez", "foldChange", "padj")], padj < 0.05 & : error in evaluating the argument 'x' in selecting a method for function 'subset': Error: object 'genego' not found
```

```r
write.xlsx(genego_sig, file=paste(tissue,"genego.xls",sep="."), row.names=FALSE)
```

```
## Error in is.data.frame(x): object 'genego_sig' not found
```


```r
#> names(res)
# [1] "gene"                 "entrez.human"         "entrezgene"           "external_gene_name"  
# [5] "cardiomyopathy_score" "baseMean"             "baseMeanA"            "baseMeanB"           
# [9] "foldChange"           "log2FoldChange"       "pval"                 "padj"                
res %>% select(gene=external_gene_name,ensid=gene,baseMean,bNorm=baseMeanA,bRB=baseMeanB,fc=foldChange,log2fc=log2FoldChange,pval,padj) %>% filter(padj<myPval) %>% arrange(pval)->resSig
```

```
## Error in eval(expr, envir, enclos): object 'res' not found
```

```r
datatable(resSig, caption=paste("Significant (P<",myPval,") differential expression in ANT1 after the correction for multiple testing.",nrow(resSig),"genes."))
```

```
## Error in base::rownames(data): object 'resSig' not found
```

## Upregulated genes in heart

```
## Error in head(subset(resSig[order(-resSig$fc), ], fc > 1), n = 50): error in evaluating the argument 'x' in selecting a method for function 'head': Error in subset(resSig[order(-resSig$fc), ], fc > 1) : 
##   error in evaluating the argument 'x' in selecting a method for function 'subset': Error: object 'resSig' not found
```

```
## Error in nrow(upRegs): error in evaluating the argument 'x' in selecting a method for function 'nrow': Error: object 'upRegs' not found
```

## Downregulated genes in heart

```
## Error in head(subset(resSig[order(resSig$fc), ], fc <= 1), n = 50): error in evaluating the argument 'x' in selecting a method for function 'head': Error in subset(resSig[order(resSig$fc), ], fc <= 1) : 
##   error in evaluating the argument 'x' in selecting a method for function 'subset': Error: object 'resSig' not found
```

```
## Error in nrow(downRegs): error in evaluating the argument 'x' in selecting a method for function 'nrow': Error: object 'downRegs' not found
```

## Fold Change and Significance heart
Genes with high fold change may not be significantly differentially expressed simply due to high variance.


```
## Error in eval(expr, envir, enclos): object 'res' not found
```

# Eucludian distances
![plot of chunk euclidheat]({{SLINK}}/euclidheat-1.png) 

# Session info
The R session information (including the OS info, R version and all
packages used):


```
## R version 3.2.2 (2015-08-14)
## Platform: x86_64-redhat-linux-gnu (64-bit)
## Running under: Red Hat Enterprise Linux Server release 6.7 (Santiago)
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8          LC_NUMERIC=C                 
##  [3] LC_TIME=en_US.UTF-8           LC_COLLATE=en_US.UTF-8       
##  [5] LC_MONETARY=en_US.UTF-8       LC_MESSAGES=en_US.UTF-8      
##  [7] LC_PAPER=en_US.UTF-8          LC_NAME=en_US.UTF-8          
##  [9] LC_ADDRESS=en_US.UTF-8        LC_TELEPHONE=en_US.UTF-8     
## [11] LC_MEASUREMENT=en_US.UTF-8    LC_IDENTIFICATION=en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    methods   stats     graphics  grDevices utils    
## [8] datasets  base     
## 
## other attached packages:
##  [1] xlsx_0.5.7           xlsxjars_0.6.1       rJava_0.9-7         
##  [4] DT_0.1               biomaRt_2.24.1       org.Mm.eg.db_3.1.2  
##  [7] RSQLite_1.0.0        DBI_0.3.1            AnnotationDbi_1.30.1
## [10] plyr_1.8.3           ggplot2_1.0.1        DESeq_1.20.0        
## [13] lattice_0.20-33      locfit_1.5-9.1       Biobase_2.28.0      
## [16] reshape_0.8.5        tidyr_0.3.1          dplyr_0.4.3         
## [19] stringr_1.0.0        Rsamtools_1.20.5     Biostrings_2.36.4   
## [22] XVector_0.8.0        GenomicRanges_1.20.8 GenomeInfoDb_1.4.3  
## [25] IRanges_2.2.9        S4Vectors_0.6.6      BiocGenerics_0.14.0 
## [28] knitr_1.11          
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.1        formatR_1.2.1      RColorBrewer_1.1-2
##  [4] bitops_1.0-6       tools_3.2.2        zlibbioc_1.14.0   
##  [7] digest_0.6.8       jsonlite_0.9.17    annotate_1.46.1   
## [10] evaluate_0.8       gtable_0.1.2       yaml_2.1.13       
## [13] proto_0.3-10       genefilter_1.50.0  htmlwidgets_0.5.1 
## [16] grid_3.2.2         R6_2.1.1           XML_3.98-1.3      
## [19] survival_2.38-3    reshape2_1.4.1     geneplotter_1.46.0
## [22] magrittr_1.5       htmltools_0.2.6    MASS_7.3-44       
## [25] scales_0.3.0       splines_3.2.2      assertthat_0.1    
## [28] colorspace_1.2-6   xtable_1.7-4       stringi_0.5-5     
## [31] RCurl_1.95-4.7     munsell_0.4.2
```

```
## [1] "2015-10-24 15:51:37 EDT"
```



Git Commit:
3c0b2da65baad0a330b21a4458a861954082ea63
